tests/integration_test_net.go

```diff
@@ -169,6 +177,66 @@ type integrationTestNode struct {
 	done      <-chan struct{}
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// Memory profiler.
+// if enabled with the `-test.heap.profile` flag, it will write a heap dump to
+// the `../build/profile/` directory at the end of the test run.
+// The file will be named `mem_<test_name>.pprof` where `<test_name>` is the name
+// of the test that started the profiling.
+////////////////////////////////////////////////////////////////////////////////
+
+type netProfiler struct {
+	tb testing.TB
+	// highest memory usage seen so far,
+	// used to write only the peak consumption to a file
+	highestSeen uint64
+}
+
+// since `go test` already parses the flag, this variable needs to be declared
+// at the package level, so that it can be parsed when go test parses other flags.
+var testProfileEnabled = flag.Bool("test.heap.profile", false,
+	`Enable memory profiling for integration tests. 
+	If set, a memory profile will be written to the build/profile directory 
+	at the end of the test run.`)
+
+func profileRoutine(ctx context.Context, n *netProfiler) {
```

Comment:
Identifiers should not include information about what it is identifying. You would not name something a "AddressStruct" or a "counterVariable". So this should also not be a "profileRoutine". Furthermore, functions should start with a verb. Please find a better name for this.